### PR TITLE
2DContext: Set up tests for new signature of imageData functions

### PR DESCRIPTION
**Pull request by fserb**
{get,put,create}imageData() functions are reverting from double to long.
Added/cleaned up tests for dealing with float inputs.

Relevant discussion: https://github.com/whatwg/html/issues/2433

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/213157435)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 7f3069cabf7fb7c4fab2a7c037b73a740b6f8d44
Using browser at version BuildID 20170319110229; SourceStamp e1576dd8bd9d3a4ca418cf347133b8a4957ddeca
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>4 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5180/2dcontext/pixel-manipulation/2d.imageData.create2.double.html">/2dcontext/pixel-manipulation/2d.imageData.create2.double.html</a></summary>


|                       Subtest                       | Results | Messages |
|-----------------------------------------------------|---------|----------|
|                                                     | OK      |          |
| `createImageData(w, h) double is converted to long` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5180/2dcontext/pixel-manipulation/2d.imageData.create2.zero.html">/2dcontext/pixel-manipulation/2d.imageData.create2.zero.html</a></summary>


|                             Subtest                             | Results |                                         Messages                                        |
|-----------------------------------------------------------------|---------|-----------------------------------------------------------------------------------------|
|                                                                 | OK      |                                                                                         |
| `createImageData(sw, sh) throws INDEX_SIZE_ERR if size is zero` | FAIL    | `assert_throws: function "function() { ctx.createImageData(0.99, 10); }" did not throw` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5180/2dcontext/pixel-manipulation/2d.imageData.get.double.html">/2dcontext/pixel-manipulation/2d.imageData.get.double.html</a></summary>


|                       Subtest                       | Results | Messages |
|-----------------------------------------------------|---------|----------|
|                                                     | OK      |          |
| `createImageData(w, h) double is converted to long` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5180/2dcontext/pixel-manipulation/2d.imageData.get.zero.html">/2dcontext/pixel-manipulation/2d.imageData.get.zero.html</a></summary>


|                        Subtest                         | Results |                                          Messages                                         |
|--------------------------------------------------------|---------|-------------------------------------------------------------------------------------------|
|                                                        | OK      |                                                                                           |
| `getImageData() throws INDEX_SIZE_ERR if size is zero` | FAIL    | `assert_throws: function "function() { ctx.getImageData(1, 1, 0.1, 10); }" did not throw` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/213157436)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 7f3069cabf7fb7c4fab2a7c037b73a740b6f8d44
Using browser at version 59.0.3043.0 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>4 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5180/2dcontext/pixel-manipulation/2d.imageData.create2.double.html">/2dcontext/pixel-manipulation/2d.imageData.create2.double.html</a></summary>


|                       Subtest                       | Results |                                               Messages                                              |
|-----------------------------------------------------|---------|-----------------------------------------------------------------------------------------------------|
|                                                     | OK      |                                                                                                     |
| `createImageData(w, h) double is converted to long` | FAIL    | `assert_equals: imgdata1.width === 10 (got 11[number], expected 10[number]) expected 10 but got 11` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5180/2dcontext/pixel-manipulation/2d.imageData.create2.zero.html">/2dcontext/pixel-manipulation/2d.imageData.create2.zero.html</a></summary>


|                             Subtest                             | Results |                                         Messages                                         |
|-----------------------------------------------------------------|---------|------------------------------------------------------------------------------------------|
|                                                                 | OK      |                                                                                          |
| `createImageData(sw, sh) throws INDEX_SIZE_ERR if size is zero` | FAIL    | `assert_throws: function "function () { ctx.createImageData(0.99, 10); }" did not throw` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5180/2dcontext/pixel-manipulation/2d.imageData.get.double.html">/2dcontext/pixel-manipulation/2d.imageData.get.double.html</a></summary>


|                       Subtest                       | Results |                                               Messages                                              |
|-----------------------------------------------------|---------|-----------------------------------------------------------------------------------------------------|
|                                                     | OK      |                                                                                                     |
| `createImageData(w, h) double is converted to long` | FAIL    | `assert_equals: imgdata1.width === 10 (got 11[number], expected 10[number]) expected 10 but got 11` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5180/2dcontext/pixel-manipulation/2d.imageData.get.zero.html">/2dcontext/pixel-manipulation/2d.imageData.get.zero.html</a></summary>


|                        Subtest                         | Results |                                          Messages                                          |
|--------------------------------------------------------|---------|--------------------------------------------------------------------------------------------|
|                                                        | OK      |                                                                                            |
| `getImageData() throws INDEX_SIZE_ERR if size is zero` | FAIL    | `assert_throws: function "function () { ctx.getImageData(1, 1, 0.1, 10); }" did not throw` |

</details>
</details>

File: 2dcontext/pixel-manipulation/2d.imageData.create2.double.html [hunk 1]
```diff
@@ -1,28 +1,30 @@
 <!DOCTYPE html>
 <!-- DO NOT EDIT! This test has been generated by tools/gentest.py. -->
```
Comment (annevk) [inline 2dcontext/pixel-manipulation/2d.imageData.create2.double.html]: I don't see an update to this tool. Am I missing something or is this direct modification of these tests the wrong way to go?

Comment (gsnedders) [inline 2dcontext/pixel-manipulation/2d.imageData.create2.double.html]: they come from `tests2d.yaml`

Comment (gsnedders) [inline 2dcontext/pixel-manipulation/2d.imageData.create2.double.html]: also note we'd get a CI failure if it was modifying only the generated files

Comment (annevk) [inline 2dcontext/pixel-manipulation/2d.imageData.create2.double.html]: Thanks!

**Pull request merged**